src/v/serde/rw/tags.h

```diff
@@ -0,0 +1,34 @@
+// Copyright 2023 Redpanda Data, Inc.
+//
+// Use of this software is governed by the Business Source License
+// included in the file licenses/BSL.md
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0
+
+#pragma once
+
+#include "serde/rw/tag_invoke.h"
+
+class iobuf;
+class iobuf_parser;
+
+namespace serde {
+
+inline constexpr struct write_fn {
+    template<typename T>
+    void operator()(iobuf& b, T&& x) const {
+        return tag_invoke(*this, b, std::forward<T>(x));
+    }
+} w{};
+
+inline constexpr struct read_fn {
+    template<typename T>
+    void operator()(
+      iobuf_parser& in, T& t, std::size_t const bytes_left_limit) const {
+        return tag_invoke(*this, in, t, bytes_left_limit);
+    }
+} r{};
```

Comment:
IIUC these are the functions that user types need to customize for serde support. Should these have more descriptive names? We already have a few similarly-named functions (read_*) and it is kind of hard to track what are their responsibilities and which ones are customizable. Adding just `r` to this family is a bit too much for me :)

Reply:
You would never overload `r` (since it's a variable, you would not be able to overload it anyway). This is just a tag. Your implementation would look like `void tag_invoke(tag_t<r>, ...) { /* your impl */ }`. And if you would call it directly (not via `serde::read`, it would be `r(...)`. Calling it `read` is not possible.

Do you think `read_tag` is better? Any proposals on how to call it?

Reply:
Yeah if "tag" is established terminology, read/write_tag look good to me.